hikari/impl/event_manager_base.py

```diff
@@ -71,23 +74,50 @@ class EventManagerBase(event_dispatcher.EventDispatcher):
     is the raw event name being dispatched in lower-case.
     """
 
-    __slots__: typing.Sequence[str] = ("_app", "_intents", "_listeners", "_waiters")
+    __slots__: typing.Sequence[str] = ("_app", "_consumers", "_intents", "_listeners", "_waiters")
 
     def __init__(self, app: traits.BotAware, intents: intents_.Intents) -> None:
         self._app = app
+        self._consumers: typing.Dict[str, typing.List[ConsumerT]] = {}
         self._intents = intents
         self._listeners: ListenerMapT[base_events.Event] = {}
         self._waiters: WaiterMapT[base_events.Event] = {}
 
+        for name, member in inspect.getmembers(self):
+            if name.startswith("on_"):
+                self.add_raw_consumer(name[3:], member)
+
+    def add_raw_consumer(self, name: str, consumer: ConsumerT, /) -> None:
+        name = name.casefold()
+        if name not in self._consumers:
+            self._consumers[name] = []
+
+        self._consumers[name].append(consumer)
+
+    def get_raw_consumers(self, name: str, /) -> typing.Sequence[ConsumerT]:
+        if consumers := self._consumers.get(name.casefold()):
+            return consumers.copy()
+
+        return []
+
+    def remove_raw_consumer(self, name: str, consumer: ConsumerT, /) -> None:
+        name = name.casefold()
+        if name not in self._consumers or consumer not in self._consumers[name]:
+            raise LookupError(name, consumer)
+
+        self._consumers[name].remove(consumer)
+
```

Comment:
What could these two be used for? With the implementation you have, we could just make everything private and not provide an interface for it

Reply:
well remove_raw_consumer is a consistency thing, since this thing isn't designed to be randomly replaced it'd make more sense for them to be able to remove consumers as well as add them.
get_raw_consumers exists for the sake of not leaving this as a black box which you work out what it's doing seeing as the fact that it loads raw consumers from attributes in itself is impl detail rather than a part of the interface. I doubt it'd be overly helpful but it's nicer than a black box